digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -139,9 +139,10 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
       withTiming(
         Await.result(combinedFuture, Duration.Inf), "End fetch vvh road links in %.3f sec"
       )
+    val roadsWithEndDate = addresses.values.flatten.filter(a => a.endDate.isDefined).map(_.linkId).toSeq
```

Comment:
Could be fetched from the database separately with a query or could use .partition to split the addresses data into two separate values.
val (roadsWithoutEndDate, roadsWithEndDate) = addresses.values.flatten.partition(_.endDate.isDefined).map(_.linkId)

Reply:
I cannot make this change on the query, because I need to filter the ones from VVH Roadlinks also. If I make the change on the query, instead of disappear, we just have 'missingRoadAddresses' on the map. I'll use the partitioner instead